src/sieve.py

```diff
@@ -51,3 +51,29 @@ def sieve3(n):
     return primes
 
 print(sieve3(15))
+
+# Even faster version.
+
+def sieve4(n):
+    candidates = [False]*2 + [True]*(n-1)
+    i = 2
+    while i*i < n:
+        if not candidates[i]: # If candidate[0] is False, not candidate[0] is 
+            # True and the if-body is executed. 
+            break # break while-loop.
+        for j in range(i**2, n+1, i): # nb i**2 must be used, not i^2.
```

Comment:
Yes, `^` is the XOR operator in Python. (In R it is exponentiation, but in most languages it is XOR).